glass/observations.py

```diff
@@ -328,3 +330,200 @@ def tomo_nz_gausserr(
+        Shell redshift limits.
+    """
+
+    def __init__(
+        self, vardepth_map: NDArray[np.float64], n_bins: int, zbins: NDArray[np.float64]
+    ) -> None:
+        self.vardepth_map = vardepth_map
+        self.n_bins = n_bins
+        self.zbins = zbins
+
+    def check_index(self, index: tuple[int, int]) -> None:
+        r"""Check the index for validity."""
+        if not isinstance(index, tuple):
+            raise TypeError("Index must be an tuple of two integers")
+
+        if index[0] >= self.n_bins:
+            raise ValueError("Leading index cannot exceed number of tomographic bins")
+
+        if index[1] >= len(self.zbins):
+            raise ValueError("Trailing index cannot exceed number of shells")
+
+    def __getitem__(self, index: tuple[int, int]) -> NDArray[np.float64]:
+        r"""Get the mask for the given index.
+
+        Parameters
+        ----------
+        index
+            Indices of the tomographic bin and shell pair.
+
+        Returns
+        -------
+        mask
+            Mask for the given index.
+
+        Raises
+        ------
+        ValueError
+            If the index is invalid.
+
+        """
+        self.check_index(index)
+
+        return self.vardepth_map[index[0]]
+
+
+class AngularLosVariableDepthMask(AngularVariableDepthMask):
+    r"""Variable depth mask for tomographic bins.
+
+    This class allows to create a mask with a different variable
+    depth mask in both the angular and the line-of-sight directions
+    for each tomographic bin.
+
+    Parameters
+    ----------
+    vardepth_map
+        Map of variable which traces the depth per tomographic bin.
+        If vardepth_tomo_functions is not provided, the values are
+        treated like a map of galaxy count ratios.
+    n_bins
+        Number of tomographic bins.
+    zbins
+        Shell redshift limits.
+    ztomo
+        Tomographic redshift bin limits.
+    dndz
+        Redshift distributions per tomographic bin.
+    z
+        Redshift domain of dndz.
+    dndz_vardepth
+        Redshift distribution affected by variable depth
+        (n_bins x len(vardepth_values) x len(z)).
+    vardepth_values
+        Variable depth tracer domain/values of dndz_vardepth.
+    vardepth_los_tracer
+        Map of the variable depth tracer for line-of-sight direction. If
+        provided, it is assumed to cover the same domain as vardepth_values.
+    vardepth_tomo_functions
+        List of functions which map the input vardepth_map to the ratio
+        between the galaxy count due to the variable depth and the galaxy
+        count without variable depth (for each tomographic bin). If
+        provided, it is assumed that there is one vardepth_map which
+        traces the variable depth for all tomographic bins.
+    """
+
+    def __init__(
+        self,
+        vardepth_map: NDArray[np.float64],
+        n_bins: int,
+        zbins: list(tuple[float, float]),
+        ztomo: list(tuple[float, float]),
```

Comment:
Types need square brackets
```suggestion
        zbins: list[tuple[float, float]],
        ztomo: list[tuple[float, float]],
```